### PR TITLE
Fix error usage, add samples for failure scenarios

### PR DESCRIPTION
**Pull request by rajkumarpalani**
This PR addresses the following,

1. There were 2 error representation `PayoutsError` and `Error` both being identical with minor differences. Promote usage of `Error` and remove `PayoutsError` to avoid confusion. Replaced references of `PayoutError` with `Error`
2. `CreatePayoutResponse` and `PayoutBatch` has references to `PayoutError` which is unused/not required. Removed those references
3. `PayoutBatchHeader` and `PayoutHeader` has references to `Error` which is unused/not required. Removed those references
4.  Add `location` to `ErrorDetails`
5.  Release a 1.1.0 version for the error related changes
6. Add samples on failures and how to parse failure response entity from the `HttpException`

### CODE REVIEW

File: payouts-sdk-sample/src/main/java/com/paypal/GetPayoutBatch.java [hunk 1]
```diff
@@ -28,12 +35,19 @@
...
+        try {
+            HttpResponse<PayoutBatch> response = client().execute(request);
 
-        return response;
+            System.out.println("Response Body:");
+            System.out.println(new JSONObject(new Json().serialize(response.result())).toString(4));
+            return response;
+        } catch (HttpException e) {
+            String error = e.getMessage();
+            Error payoutError = encoder.deserializeResponse(new ByteArrayInputStream(error.getBytes(StandardCharsets.UTF_8)), Error.class, e.headers());
```
Comment (aninja8) [inline payouts-sdk-sample/src/main/java/com/paypal/GetPayoutBatch.java:46]: should we just add util for this deserialization?

**Review by aninja8 — COMMENTED** [review]
Suggested few changes

File: README.md [hunk 2]
```diff
@@ -105,6 +105,69 @@ public class CreatePayoutExample {
...
+			PayoutsPostRequest httpRequest = new PayoutsPostRequest().requestBody(request);
+			Credentials.client.execute(httpRequest);
+		} catch (IOException ioe) {
+			if (ioe instanceof HttpException) {
+				// Something went wrong server-side
+				HttpException he = (HttpException) ioe;
+                
+				// Parse failure response to get the reason for failure
+				Encoder encoder = new Encoder();
+				Error payoutError = encoder.deserializeResponse(new ByteArrayInputStream(error.getBytes(StandardCharsets.UTF_8)), Error.class, e.headers());
```
Comment (aninja8) [inline README.md:158]: where is this `error` initialized?

File: README.md [hunk 3]
```diff
@@ -105,6 +105,69 @@ public class CreatePayoutExample {
...
+                        .note("Enjoy your Payout!!")
+                        .recipientType("EMAIL"))
+                .items(items);
+
+		try {
+			// Call API with your client and get a response for your call
+			PayoutsPostRequest httpRequest = new PayoutsPostRequest().requestBody(request);
+			Credentials.client.execute(httpRequest);
+		} catch (IOException ioe) {
+			if (ioe instanceof HttpException) {
```
[the remark this answers is not in the document]
Comment (rajkumarpalani) [reply README.md:152]: HttpException is a sub class of IOException. IOException can be thrown if there is any issue with network on client side which need to be handled as well. The sample is just to showcase that

File: README.md [hunk 4]
```diff
@@ -105,6 +105,69 @@ public class CreatePayoutExample {
...
+public class CreatePayoutExample {
+	public static void main(String[] args) {
+
+		// Construct a request object and set desired parameters
+		// Here, CreatePayoutRequest() creates a POST request to /v1/payments/payouts
+		List<PayoutItem> items = IntStream
+                .range(1, 6)
+                .mapToObj(index -> new PayoutItem()
+                        .senderItemId("Test_txn_" + index)
+                        .note("Your 5$ Payout!")
```
[the remark this answers is not in the document]
Comment (rajkumarpalani) [reply README.md:131]: Fixed

File: README.md (hunk 2 above)
Comment (rajkumarpalani) [reply README.md:158]: Fixed

File: payouts-sdk-sample/src/main/java/com/paypal/GetPayoutBatch.java (hunk 1 above)
Comment (rajkumarpalani) [reply payouts-sdk-sample/src/main/java/com/paypal/GetPayoutBatch.java:46]: Encoder is also an utility to deserialize, adding another util would just be a wrapper over Encoder

**Review by aninja8 — APPROVED** [review]
LGTM! Travis failed though. That has to be checked out.

File: payouts-sdk/src/main/java/com/paypal/payouts/ErrorDetails.java [hunk 5]
```diff
@@ -50,4 +50,20 @@ public ErrorDetails issue(String issue) {
         this.issue = issue;
         return this;
     }
+
+    /**
+     * REQUIRED
+     * The reason for the error.
```
Comment (aninja8) [inline payouts-sdk/src/main/java/com/paypal/payouts/ErrorDetails.java:56]: should be location right ?

Comment (rajkumarpalani) [reply payouts-sdk/src/main/java/com/paypal/payouts/ErrorDetails.java]: fixed

**Pull request merged**